deepwell/src/services/filter/service.rs

```diff
@@ -91,6 +99,33 @@ impl FilterService {
             ..Default::default()
         };
 
+        // Handle case-sensitivity logic
+        if let ProvidedValue::Set(case_sensitive) = case_sensitive {
+            match regex {
+                // If the regex is being changed, add case-insensitivity flag if case-insensitive.
+                ProvidedValue::Set(ref mut regex) if !case_sensitive => {
+                    regex.insert_str(0, "(?i)")
+                }
+
+                // If the regex is not being changed, remove (and conditionally readd) the
+                // case-insensitivity flag from the database's regex.
+                ProvidedValue::Unset => {
+                    let mut model_regex = str!(model.get(filter::Column::Regex).as_ref());
+
+                    model_regex = str!(model_regex.trim_start_matches("(?i)"));
```

Comment:
Instead of re-allocating, since you already get an owned string on line 113, you can instead do:
```suggestion
                    if model_regex.starts_with("(?i)") {
                        // SAFETY: In this branch we know the string starts with 4 ASCII characters
                        model_regex.drain(..4);
                    }
```

Reply:
I don't know if this is necessarily better. I feel that `trim_start_matches()` is more understandable than `drain(..4)`. Additionally I don't know if this saves much on computation time, but I don't have any hard statistics on this. I also think it's needlessly verbose comparatively, and also doesn't strip the prefix multiple times if present like `trim_start_matches` does.

Reply:
Okay I'll add it as a helper function instead then.